### ISSUE TITLE
QSelect scrollbar not styled

### ISSUE THREAD

**Issue by meatfighter** [quasarframework/quasar#17360]
### What happened?

If the dropdown of a QSelect is too tall for its container, a scrollbar appears. But the scrollbar is not styled. It looks generic.

https://codepen.io/Michael-Birken/pen/RwzwXxv?editors=101

![image](https://github.com/quasarframework/quasar/assets/54445200/e5a193d0-ae6e-4d50-a1aa-7a59352cbfa5)

### What did you expect to happen?

The QSelect menu scrollbar should look like it does in the Quasar Framework docs:

https://quasar.dev/vue-components/select

![image](https://github.com/quasarframework/quasar/assets/54445200/550547a9-3e90-47ed-b5fe-9c87bd9c9e9e)

### Reproduction URL

https://codepen.io/Michael-Birken/pen/RwzwXxv?editors=101

### How to reproduce?

1. Go to https://codepen.io/Michael-Birken/pen/RwzwXxv?editors=101
2. Narrow the height of the browser to force the QSelect menu to manifest a scrollbar.
3. Observe the scrollbar's sytle.

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Components (quasar)

### Platforms/Browsers

Firefox, Chrome, Microsoft Edge

### Additional context

The same thing happens to a QDialog if it is too tall. A scrollbar appears on its right. But that scrollbar is not properly styled either.

https://codepen.io/Michael-Birken/pen/GRbRVQL?editors=101

![image](https://github.com/quasarframework/quasar/assets/54445200/900f3426-a1ae-4c87-8e4f-4ade5aade3c4)

**Comment by RamonDonadeu** [comment]
That is because the quasar docs have styled the scrollbar.


https://codepen.io/Michael-Birken/pen/RwzwXxv

**Comment by meatfighter** [comment]
> That is because the quasar docs have styled the scrollbar.
> 
> https://codepen.io/Michael-Birken/pen/RwzwXxv

Why doesn't Quasar do that? How did the docs style it? Can I add that to my Quasar project in a way that works across browsers?

**Comment by RamonDonadeu** [comment]
https://www.w3schools.com/howto/howto_css_custom_scrollbar.asp

Check this webpage for Chrome, Edge, Safari and Opera.
For firefox it changes but you can find it easy probably

**Comment by RamonDonadeu** [comment]
Oops i just realized that i didn't paste my codepen

https://codepen.io/RamonDonadeu/pen/GRbpQMa

> That is because the quasar docs have styled the scrollbar.
> 
> https://codepen.io/Michael-Birken/pen/RwzwXxv

**Comment by meatfighter** [comment]
> Oops i just realized that i didn't paste my codepen
> 
> https://codepen.io/RamonDonadeu/pen/GRbpQMa
> 
> > That is because the quasar docs have styled the scrollbar.
> > https://codepen.io/Michael-Birken/pen/RwzwXxv

Thanks for suggestions. But I don't think that works across browsers, particularly Firefox. More importantly, Quasar should do this out of the box, which is why I filed this bug report.

**Comment by rstoenescu** [comment]
This has to be styled in devland by the developers themselves. We cannot and should not interfere at such level with the CSS from the framework level.

**Comment by LuSrackhall** [comment]
> 这必须由开发人员自己在 devland 中设置样式。我们不能也不应该从框架级别干预 CSS 的这种级别。

In fact, I also think this is indeed a problem because the q-select component does not provide a way to define the position of the corresponding menu or dialog, which makes CSS operations on them very inconvenient.

Although the component should not interfere with how users define scrolling styles, the location where these styles are written is a problem. Or, if the position of the menu or dialog in q-select could be defined, it would allow users to easily define the styles they want for the dropdown menu list or dialog list.

Furthermore, for the dropdown list of the menu or dialog in q-select, it uses Quasar's virtual-scroll by default. Would it be possible to allow users to customize this part, for instance, by letting users choose a third-party virtual-scroll library to apply to them instead of being forced to use the default virtual-scroll?

In short, in the usage of the q-select component, there is no way to provide a higher degree of customization for the entire menu dropdown box or dialog dropdown box. If this part could be separated for users to use, that would be great. (It seems that it cannot be achieved through slots, as many default behaviors of q-select need to be bound to it.)
